### PR TITLE
chore: Universal Linking for Android

### LINKED ISSUE #786: Enhancement: Setup apple-app-site-association file for universal linking
Working on [chatwoot-mobile-app/issues/141](https://github.com/chatwoot/chatwoot-mobile-app/issues/141).

In order to work Universal Linking  (Allow app to launch from a link within an email or text message,) in iOS, we need some kind of association configuration in the server. 
Universal links work through a relationship between the app’s capabilities and your server apple-app-site-association file.

**Reference**

-  https://medium.com/@parkerdan/react-native-universal-linking-ios-deep-linking-android-with-a-rails-server-d993649ae6be
- http://developer.apple.com/library/archive/documentation/General/Conceptual/AppSearch/UniversalLinks.html#//apple_ref/doc/uid/TP40016308-CH12-SW1
- https://medium.com/hackernoon/react-native-deep-linking-for-ios-and-android-d33abfba7ef3
- https://branch.io/resources/aasa-validator/
- https://blog.voom.flights/universal-links-with-react-native-ios-18457481ccaf

### PR DESCRIPTION
**Pull request by aniravi24**
# Pull Request

## Description

Sorry I didn't make a separate issue for this, as it ties to universal linking for iOS. This is the universal linking config for Android, so that links to your hosted chatwoot directly go into the app.

Note that this will cause all links to be intercepted unless you specifically indicate on the client side manifest XML that you want routes such as /conversations to go through the app.

Fixes #786

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

(Not sure if adding environment variables that aren't used is a breaking change)

## How Has This Been Tested?

This has been tested in production with a self-hosted chatwoot instance. You have to deploy this wherever the app is pointing to in its linking config, and then open a link that points to that domain. It will automatically go inside the app instead of prompting you at the OS level as to which app to open it with.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by sojan-official** [comment]
@aniravi24  could you add specs for the new route as well?  
example ref: https://github.com/chatwoot/chatwoot/blob/develop/spec/controllers/apple_app_spec.rb

**Comment by aniravi24** [comment]
> @aniravi24 could you add specs for the new route as well?
> example ref: https://github.com/chatwoot/chatwoot/blob/develop/spec/controllers/apple_app_spec.rb

Sure, sorry I missed that, first time doing ruby!

File: .env.example [hunk 1]
```diff
@@ -116,6 +116,10 @@ SLACK_CLIENT_SECRET=
 ### Change this env variable only if you are using a custom build mobile app
 ## Mobile app env variables
 IOS_APP_ID=6C953F3RX2.com.chatwoot.app
+ANDROID_BUNDLE_ID=com.chatwoot.app
+
+# https://developers.google.com/android/guides/client-auth (use keytool to print the fingerprint in the first section)
+ANDROID_SHA256_CERT_FINGERPRINT=05:A2:2C:35:EE:F2:51:23:72:4D:72:67:A5:6C:8C:58:22:2A:00:D6:DB:F6:45:D5:C1:82:D2:80:A4:69:A8:FE
```
Comment (pranavrajs) [inline .env.example:122]: Should this be committed to the repo? @muhsin-k Can you please take a look?

**Review by pranavrajs — COMMENTED** [review]
Nice work! I have a comment about committing SHA_CERT_FINGERPRINT

File: .env.example (hunk 1 above)
Comment (aniravi24) [reply .env.example:122]: This is just a dummy example from google's docs!

**Comment by aniravi24** [comment]
> Nice work! I have a comment about committing SHA_CERT_FINGERPRINT

responded above :smile:

**Comment by muhsin-k** [comment]
@aniravi24 Good work! We will merge this PR once test it out in the testing environment.

**Review by muhsin-k — APPROVED** [review]

**Review by muhsin-k — DISMISSED** [review]

**Review by pranavrajs — DISMISSED** [review]

**Review by sojan-official — APPROVED** [review]

**Review by muhsin-k — APPROVED** [review]

**Pull request merged**